File: solutions/python/resistor-color-duo/1/resistor_color_duo.py
'''Returns band color value for first 2 colors passed as argument'''

color_encodings = ["black", "brown", "red", "orange", "yellow", "green", "blue", "violet", "grey", "white"]

def value(colors):
    encoding = ""
    count = 0

    for color in colors:
        count += 1
        if count == 3:
            break
        encoding = encoding + str(color_encodings.index(color))

    return int(encoding)
    

File: solutions/cpp/resistor-color-duo/1/resistor_color_duo.cpp
#include "resistor_color_duo.h"

#include <iostream>

namespace resistor_color_duo {

// TODO: add your solution here
    int value(std::vector<string> input) {
        std::map<std::string, int> colorMap = {
            {"black", 0},
            {"brown", 1},
            {"red", 2},
            {"orange", 3},
            {"yellow", 4},
            {"green", 5},
            {"blue", 6},
            {"violet", 7},
            {"grey", 8},
            {"white", 9}
        };

        string output{};
        
        for (int i = 0; i < 2; i++) {
            output += std::to_string(colorMap[input[i]]);
        }

        return std::stoi(output);
    };
}  // namespace resistor_color_duo
